src/components/SingleArticle.jsx

```diff
@@ -61,6 +62,37 @@ const SingleArticle = () => {
         />
     </div>;
 
+    const handlerArticleVoteUp = () => {
+        setVoteError(false)
+        setArticle(article => ({
+            ...article,
+            votes: article.votes + 1
+        })) 
+        patchArticleVote(article.article_id, 1).catch((err) => {
+            setArticle(article => ({
+                ...article,
+                votes: article.votes - 1
+            }))
+            setVoteError(true)
+        })
+    }
+
+
+    const handlerArticleVoteDown = () => {
+        setVoteError(false)
+        setArticle(article => ({
+            ...article,
+            votes: article.votes - 1
+        }))
+        patchArticleVote(article.article_id, -1).catch((err) => {
+            setArticle(article => ({
+                ...article,
+                votes: article.votes + 1
+            }))
+            setVoteError(true)
```

Comment:
Nice to see this here. When you get to the error handling ticket I think it might be worth a revisit. It is very clear that something has gone wrong when the whole page disappears to a 404 😄 but perhaps you could leave the user on the article still with a nice message.